File: partiql-ast/src/main/java/org/partiql/ast/v1/DropTable.java
package org.partiql.ast.v1;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * TODO docs, equals, hashcode
 */
public class DropTable extends DDL {
    @NotNull
    public Identifier table;

    public DropTable(@NotNull Identifier table) {
        this.table = table;
    }

    @NotNull
    @Override
    public Collection<AstNode> children() {
        List<AstNode> kids = new ArrayList<>();
        kids.add(table);
        return kids;
    }

    @Override
    public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
        return visitor.visitDropTable(this, ctx);
    }
}

File: partiql-ast/src/main/kotlin/org/partiql/ast/v1/DropTable.kt
package org.partiql.ast.v1

/**
 * TODO docs, equals, hashcode
 */
public class DropTable(
    @JvmField
    public var table: Identifier,
) : DDL() {
    public override fun children(): Collection<AstNode> {
        val kids = mutableListOf<AstNode?>()
        kids.add(table)
        return kids.filterNotNull()
    }

    public override fun <R, C> accept(visitor: AstVisitor<R, C>, ctx: C): R =
        visitor.visitDropTable(this, ctx)
}
